### PR TITLE
[FEATURE] SDK - Add support for response status

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR adds support to provide the status for record responses. This is quite important to admin users when restoring annotated data from external sources. Also, the migration script from the legacy dataset can benefit the change.

When several statuses are found for the same user before sending data to the server, a warning is shown to the users and the `draft` status is selected.

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-sdk/src/argilla_sdk/responses.py [hunk 1]
```diff
@@ -49,10 +59,14 @@ def __init__(
         if user_id is None:
             raise ValueError("user_id is required")
 
+        if isinstance(status, str):
```
Comment (burtenshaw) [inline argilla-sdk/src/argilla_sdk/responses.py:62]: Should we warn the user when they create responses with a status of submitted and a different user_id?

**Review by burtenshaw — COMMENTED** [review]
Just a low comment on warning.

File: argilla-sdk/src/argilla_sdk/responses.py (hunk 1 above)
Comment (frascuchon) [reply argilla-sdk/src/argilla_sdk/responses.py:62]: Not sure to understand your question. Do you mean setting a different user that connected one? I can see some migration scenarios where an admin user wants to provide several responses from different users. So, in that case makes sense to provide different user IDs. But again, not sure to understand your question correctly.

Comment (davidberenstein1957) [reply argilla-sdk/src/argilla_sdk/responses.py:62]: @burtenshaw I would say no. I think people should consciously change these kinds of things and probably know what they are doing in that case.

File: argilla-sdk/src/argilla_sdk/responses.py [hunk 2]
```diff
@@ -171,16 +184,34 @@ def to_dict(self) -> Dict[str, Any]:
...
+        statuses = set([answer.status for answer in answers if answer.status is not None])
+        if len(statuses) > 1:
+            warnings.warn(f"Multiple status found in user answers. Using {ResponseStatus.draft!r} as default.")
+        elif len(statuses) == 1:
+            return ResponseStatusModel(next(iter(statuses)))
+        return ResponseStatusModel.draft
+
+    def _compute_user_id_from_answers(self, answers: List[Response]) -> UUID:
+        user_ids = set([answer.user_id for answer in answers if answer.user_id is not None])
+        if len(user_ids) == 0:
```
Comment (davidberenstein1957) [inline argilla-sdk/src/argilla_sdk/responses.py]: weren't we going to set the user ID to the client one by default? for me this makes the most sense in workflows where an AI engineer decides to add new content and might quickly want to register responses,.

Comment (davidberenstein1957) [reply argilla-sdk/src/argilla_sdk/responses.py]: Potentially a warning could fit here.

Comment (frascuchon) [reply argilla-sdk/src/argilla_sdk/responses.py]: In deed, this condition won't happen. I will remove it.

File: argilla-sdk/src/argilla_sdk/responses.py (hunk 1 above)
Comment (frascuchon) [reply argilla-sdk/src/argilla_sdk/responses.py:62]: this feat. is an admin one. I agree with @davidberenstein1957

**Review by burtenshaw — APPROVED** [review]

File: argilla-sdk/tests/integration/test_add_records.py [hunk 3]
```diff
@@ -605,17 +604,17 @@ def test_add_records_objects_with_responses(client: Argilla):
     records = [
```
Comment (davidberenstein1957) [inline argilla-sdk/tests/integration/test_add_records.py:604]: should we include a test wihout status?

Comment (frascuchon) [reply argilla-sdk/tests/integration/test_add_records.py:604]: To check the default value, right?

Comment (davidberenstein1957) [reply argilla-sdk/tests/integration/test_add_records.py:604]: Yes

Comment (frascuchon) [reply argilla-sdk/tests/integration/test_add_records.py:604]: Done

**Review by davidberenstein1957 — APPROVED** [review]

**Pull request merged**